lte/cloud/go/services/subscriberdb/storage/subscribers.go

```diff
@@ -0,0 +1,302 @@
+ LICENSE file in the root directory of this source tree.
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package storage
+
+import (
+	"database/sql"
+	"fmt"
+
+	lte_protos "magma/lte/cloud/go/protos"
+	configurator_storage "magma/orc8r/cloud/go/services/configurator/storage"
+	"magma/orc8r/cloud/go/sqorc"
+
+	"github.com/Masterminds/squirrel"
+	"github.com/golang/protobuf/proto"
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+)
+
+const (
+	subscriberTableName    = "subscribers"
+	subscriberTmpTableName = "subscribers_tmp"
+	subscriberNidCol       = "network_id"
+	subscriberSidCol       = "subscriber_id"
+	subscriberProtoCol     = "subscriber_proto"
+)
+
+type SubStore struct {
+	db      *sql.DB
+	builder sqorc.StatementBuilder
+}
+
+func NewSubStore(db *sql.DB, builder sqorc.StatementBuilder) *SubStore {
+	return &SubStore{db: db, builder: builder}
+}
+
+func (l *SubStore) Initialize() error {
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		_, err := l.builder.CreateTable(subscriberTableName).
+			IfNotExists().
+			Column(subscriberNidCol).Type(sqorc.ColumnTypeText).NotNull().EndColumn().
+			Column(subscriberSidCol).Type(sqorc.ColumnTypeText).NotNull().EndColumn().
+			Column(subscriberProtoCol).Type(sqorc.ColumnTypeBytes).NotNull().EndColumn().
+			PrimaryKey(subscriberNidCol, subscriberSidCol).
+			RunWith(tx).
+			Exec()
+		if err != nil {
+			return nil, errors.Wrap(err, "initialize sub proto store table")
+		}
+		_, err = l.builder.CreateTable(subscriberTmpTableName).
+			IfNotExists().
+			Column(subscriberNidCol).Type(sqorc.ColumnTypeText).NotNull().EndColumn().
+			Column(subscriberSidCol).Type(sqorc.ColumnTypeText).NotNull().EndColumn().
+			Column(subscriberProtoCol).Type(sqorc.ColumnTypeBytes).NotNull().EndColumn().
+			PrimaryKey(subscriberNidCol, subscriberSidCol).
+			RunWith(tx).
+			Exec()
+		if err != nil {
+			return nil, errors.Wrap(err, "initialize sub proto store tmp table")
+		}
+		return nil, nil
+	}
+	_, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	return err
+}
+
+// InitiateUpdate prepares the db tables for a batch update.
+func (l *SubStore) InitiateUpdate() error {
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		_, err := l.builder.
+			Delete(subscriberTmpTableName).
+			RunWith(tx).
+			Exec()
+		return nil, errors.Wrapf(err, "clear sub protos tmp table")
+	}
+	_, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	return err
+}
+
+// InsertMany inserts a batch of subscriber data into the temporary table.
+func (l *SubStore) InsertMany(network string, subProtos []*lte_protos.SubscriberData) error {
+	// NOTE: Caller of this function should guard against the case where
```

Comment:
Here and below: this is a weird place for this comment. Instead, add a note in the docstring that the caller should enforce some reasonable max size of subProtos, in accordance with its max page size